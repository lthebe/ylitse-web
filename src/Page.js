import React from 'react';
import PropTypes from 'prop-types';
import Paper from 'material-ui/Paper';
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
    root: theme.mixins.gutters({
        marginTop: theme.spacing.unit * 4,
        marginLeft: '25%',
        marginRight: '25%',
        paddingTop: 24,
        paddingBottom: 24,
    }),
});

const Page = props => (
    <Paper className={props.classes.root}>
        {props.children}
    </Paper>
);

Page.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.array,
        PropTypes.element,
        PropTypes.string,
    ]).isRequired,
    classes: PropTypes.shape({ root: PropTypes.string }).isRequired,
};

export default withStyles(styles)(Page);
