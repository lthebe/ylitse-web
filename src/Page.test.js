import React from 'react';
import { shallow } from 'enzyme';
import Page from './Page';

describe('Page', () => {
    test('renders correctly', () => {
        const wrapper = shallow(<Page><div>TEST</div></Page>).dive();

        expect(wrapper).toMatchSnapshot();
    });
});
