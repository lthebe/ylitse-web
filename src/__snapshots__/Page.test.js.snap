// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page
    classes={
      Object {
        "root": "Page-root-1",
      }
    }
  >
    <div>
      TEST
    </div>
  </Page>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        TEST
      </div>,
      "className": "Page-root-1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "TEST",
      },
      "ref": null,
      "rendered": "TEST",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          TEST
        </div>,
        "className": "Page-root-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TEST",
        },
        "ref": null,
        "rendered": "TEST",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
